tests/client/core/components/TestSearchForm.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import { Simulate, renderIntoDocument } from 'react-addons-test-utils';
+
+import * as actions from 'core/actions';
+import * as coreApi from 'core/api';
+import { SearchFormBase, mapDispatchToProps, mapStateToProps }
+  from 'core/components/Search/SearchForm';
+
+
+describe('<SearchForm />', () => {
+  const pathname = '/somewhere';
+  let api;
+  let loadAddon;
+  let router;
+  let root;
+  let form;
+  let input;
+
+  class SearchFormWrapper extends React.Component {
+    static childContextTypes = {
+      router: React.PropTypes.object,
+    }
+
+    getChildContext() {
+      return { router };
+    }
+
+    render() {
+      return (<SearchFormBase
+        pathname={pathname} api={api}
+        loadAddon={loadAddon} ref="root"
+      />);
+    }
+  }
+
+  beforeEach(() => {
+    router = { push: sinon.spy() };
+    loadAddon = sinon.stub();
+    api = sinon.stub();
+    root = renderIntoDocument(<SearchFormWrapper />).refs.root;
+    form = root.refs.form;
+    input = root.refs.query;
+  });
+
+  it('renders a form', () => {
+    assert.ok(form.classList.contains('search-form'));
+  });
+
+  it('renders a search input', () => {
+    assert.equal(input.placeholder, 'Search');
+    assert.equal(input.type, 'search');
+  });
+
+  it('does changes the URL on submit', () => {
+    assert(!router.push.called);
+    input.value = 'adblock';
+    Simulate.submit(form);
+    assert(router.push.called);
+  });
+
+  // it('does nothing on non-Enter keydowns', () => {
```

Comment:
They're in the admin (where we watch for enter rather than just making it easier with form submit), so they don't need to be (or belong) here.